shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-spring/shardingsphere-sharding-spring-boot-starter/src/test/java/org/apache/shardingsphere/sharding/spring/boot/ShardingSpringBootStarterTest.java

```diff
@@ -68,6 +74,47 @@ public void assertKeyGenerateAlgorithm() {
+        assertThat(shardingTableRuleConfigurationList.get(1).getTableShardingStrategy().getShardingAlgorithmName(), is("orderItemTableShardingAlgorithm"));
+        assertThat(shardingTableRuleConfigurationList.get(1).getKeyGenerateStrategy().getColumn(), is("order_item_id"));
+        assertThat(shardingTableRuleConfigurationList.get(1).getKeyGenerateStrategy().getKeyGeneratorName(), is("keyGenerator"));
+
+        assertThat(shardingRuleConfiguration.getBindingTableGroups().size(), is(2));
+        List<String> bindingTableGroupsList = new ArrayList<>(shardingRuleConfiguration.getBindingTableGroups());
+        assertThat(bindingTableGroupsList.get(0), is("t_order"));
+        assertThat(bindingTableGroupsList.get(1), is("t_order_item"));
+
+        assertThat(shardingRuleConfiguration.getBroadcastTables().size(), is(1));
+        assertThat(shardingRuleConfiguration.getBroadcastTables().iterator().next(), is("t_config"));
+
+        assertThat(shardingRuleConfiguration.getDefaultDatabaseShardingStrategy(), instanceOf(StandardShardingStrategyConfiguration.class));
+        assertThat(((StandardShardingStrategyConfiguration) shardingRuleConfiguration.getDefaultDatabaseShardingStrategy()).getShardingColumn(), is("user_id"));
+        assertThat(shardingRuleConfiguration.getDefaultDatabaseShardingStrategy().getShardingAlgorithmName(), is("databaseShardingAlgorithm"));
+    
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().size(), is(3));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("orderItemTableShardingAlgorithm"), instanceOf(InlineShardingAlgorithm.class));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("orderItemTableShardingAlgorithm").getType(), is("INLINE"));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("orderItemTableShardingAlgorithm").getProps().getProperty("algorithm.expression"), is("t_order_item_$->{order_id % 2}"));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("databaseShardingAlgorithm"), instanceOf(InlineShardingAlgorithm.class));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("databaseShardingAlgorithm").getType(), is("INLINE"));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("databaseShardingAlgorithm").getProps().getProperty("algorithm.expression"), is("ds_$->{user_id % 2}"));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("orderTableShardingAlgorithm"), instanceOf(InlineShardingAlgorithm.class));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("orderTableShardingAlgorithm").getType(), is("INLINE"));
+        assertThat(shardingRuleConfiguration.getShardingAlgorithms().get("orderTableShardingAlgorithm").getProps().getProperty("algorithm.expression"), is("t_order_$->{order_id % 2}"));
+
```

Comment:
Please remove useless blank line

Reply:
that's for separete the different case . is that not the ss coding rule ?

Reply:
For example:
assertShardingConfiguration can be composited by assertTableShardingStrategy, assertDatabaseShardingStrategy, assertKeyGenerator etc...